docs/sections/contributor_guide/developer_setup.rst

```diff
@@ -2,4 +2,173 @@
 Developer Setup
 ***************
 
-TBD
+Creating a development shell
+============================
+
+To create an interactive development ``bash`` shell:
+
+
+1. Download, install, and activate the latest `Miniforge3`_ for your system. If an existing conda (Miniforge, Miniconda, Anaconda, etc.) installation is available and writable, you may activate that and skip the first 5 recipe steps below.
+2. Install the `condev`_ package into the base environment.
+3. In a clone of the `workflow-tools repository`_, create the development shell.
+
+
+This recipe uses the ``aarch64`` (64-bit ARM) Miniforge for Linux, and installs into ``$HOME/conda``. Adjust as necessary for your target system.
+
+.. code::
+
+   wget https://github.com/conda-forge/miniforge/releases/latest/download/Miniforge3-Linux-aarch64.sh
+   bash Miniforge3-Linux-aarch64.sh -bfp ~/conda
+   rm Miniforge3-Linux-aarch64.sh
+   source ~/conda/etc/profile.d/conda.sh
+   conda activate
+   conda install -y -c maddenp condev
+   cd /to/your/workflow-tools/clone
+   make devshell
+
+
+.. _paragraph: 
+
+If the above is successful, you will be in a ``workflow-tools``
+development shell. See below for usage information. You may exit the
+shell with ``exit`` or ``ctrl-d``.
+
+Future ``make devshell`` invocations will be almost instantaneous, as
+the underlying virtual environment will already exist. In general, all
+source code changes will be immediately live in the development shell,
+subject to execution, test, etc. But some changes – especially to the
+contents of the ``recipe/`` directory, or to the ``src/setup.py`` module
+– may require recreation of the development shell. If you know this is
+needed, or when in doubt: Exit the development shell, run
+``conda env remove -n DEV-uwtools`` to remove the old environment, then
+run ``make devshell`` to recreate it.
+
+If your development shell misses any functionality you’re used to in
+your main shell, you can create a ``~/.condevrc`` file, which will be
+sourced by ``make devshell``. When in doubt, you might
+
+
+sourced by ``make devshell``. When in doubt, you might
+
+.. code::
+
+   cat <<EOF >~/.condevrc
+   source ~/.bashrc
+   EOF
+
+
+Using a development shell
+=========================
+
+In an active development shell, the following ``make`` targets are
+available and act on all ``.py`` files under ``src/``:
+
+
++---------------------+------------------------------------------------------------+
+| Command             |  Description                                               |
++=====================+============================================================+
+| ``make format``     | Format with ``black`` and ``isort``                        |
```

Comment:
Also, `make format` does more than this now -- it also formats docstrings and `.jsonschema` documents. Could you have a look at what the `format` target in the top-level `Makefile` does and update this description, probably including links to the other tools that are also now in use?

Reply:
Definitely updating this description, here is what `format` currently checks and formats based on the `Makefile`:
```
black src
isort src
docformatter
jsonschema
```

It also would be good to update the paragraph below this table. If relevant to the developer, we could expand on the mention of running `make format` often and include some information from [UW-364](https://github.com/ufs-community/workflow-tools/pull/338) PR for adding a check for unformatted PR code? Maybe this is reaching a bit for a connection to `make format`.

Reply:
This should be updated now.